From 7825a689bc5d227c3d594636ec06af3c7df0c280 Mon Sep 17 00:00:00 2001
From: Cole Mickens <cole.mickens@gmail.com>
Date: Thu, 26 May 2022 17:15:57 -0700
Subject: [PATCH] flakes: flake.lock location env var

---
 src/libexpr/flake/flake.cc | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/src/libexpr/flake/flake.cc b/src/libexpr/flake/flake.cc
index 35c841897..ec7f05c2b 100644
--- a/src/libexpr/flake/flake.cc
+++ b/src/libexpr/flake/flake.cc
@@ -336,8 +336,12 @@ LockedFlake lockFlake(
     try {
 
         // FIXME: symlink attack
+        char * suffix = getenv("FLAKE_LOCK");
+        if (!suffix) {
+            suffix = "flake.lock";
+        }
         auto oldLockFile = LockFile::read(
-            flake.sourceInfo->actualPath + "/" + flake.lockedRef.subdir + "/flake.lock");
+            flake.sourceInfo->actualPath + "/" + flake.lockedRef.subdir + "/" + suffix);
 
         debug("old lock file: %s", oldLockFile);
 
@@ -551,6 +555,7 @@ LockedFlake lockFlake(
                                 inputFlake.inputs, childNode, inputPath,
                                 oldLock
                                 ? std::dynamic_pointer_cast<const Node>(oldLock)
+                                // TODO: can there be self-references that would then be wrong? (maybe not they'd be circular?)
                                 : LockFile::read(
                                     inputFlake.sourceInfo->actualPath + "/" + inputFlake.lockedRef.subdir + "/flake.lock").root,
                                 oldLock ? lockRootPath : inputPath, localPath, false);
@@ -606,7 +611,11 @@ LockedFlake lockFlake(
                         if (!lockFlags.updateLockFile)
                             throw Error("flake '%s' requires lock file changes but they're not allowed due to '--no-update-lock-file'", topRef);
 
-                        auto relPath = (topRef.subdir == "" ? "" : topRef.subdir + "/") + "flake.lock";
+                        char * suffix = getenv("FLAKE_LOCK");
+                        if (!suffix) {
+                            suffix = "flake.lock";
+                        }
+                        auto relPath = (topRef.subdir == "" ? "" : topRef.subdir + "/") + suffix;
 
                         auto path = *sourcePath + "/" + relPath;
 
-- 
2.36.0

