From 4f3d3b4f33c96c4ceec8dd13568232c0ab497c82 Mon Sep 17 00:00:00 2001
From: Cole Mickens <cole.mickens@gmail.com>
Date: Wed, 10 Nov 2021 01:54:47 -0800
Subject: [PATCH] gbm

---
 render/egl.c          |  7 ++++---
 render/wlr_renderer.c | 13 +++++++++++++
 2 files changed, 17 insertions(+), 3 deletions(-)

diff --git a/render/egl.c b/render/egl.c
index ec23ce8d8..430033b20 100644
--- a/render/egl.c
+++ b/render/egl.c
@@ -139,15 +139,16 @@ static void init_dmabuf_formats(struct wlr_egl *egl) {
 		free(external_only);
 	}
 
-	char *str_formats = malloc(formats_len * 5 + 1);
+	char *str_formats = malloc(formats_len * 11 + 1);
 	if (str_formats == NULL) {
 		goto out;
 	}
 	for (int i = 0; i < formats_len; i++) {
-		snprintf(&str_formats[i*5], (formats_len - i) * 5 + 1, "%.4s ",
-			(char*)&formats[i]);
+		snprintf(&str_formats[i*11], (formats_len - i) * 11 + 1, "0x%X ",
+			(unsigned int)formats[i]);
 	}
 	wlr_log(WLR_DEBUG, "Supported DMA-BUF formats: %s", str_formats);
+	//  Supported DMA-BUF formats: 0x34324241 0x34324258 0x20203852 0x38384752 0x59565955 0x48344241 0x34325241 0x34325341 0x34325358 0x34325258 0x36314752 0x30335241 0x31303152 0x30303152 0x30334241 0x31303142 0x30303142 0x32315559 0x3231564E 0x3132564E 0x3631564E 0x3432564E 0x30313050 0x30313250 0x32313050 0x31304448 0x30444855 
 	wlr_log(WLR_DEBUG, "EGL DMA-BUF format modifiers %s",
 		has_modifiers ? "supported" : "unsupported");
 	free(str_formats);
diff --git a/render/wlr_renderer.c b/render/wlr_renderer.c
index 8a1f1e941..5ac7e3917 100644
--- a/render/wlr_renderer.c
+++ b/render/wlr_renderer.c
@@ -1,6 +1,7 @@
 #include <assert.h>
 #include <stdbool.h>
 #include <stdlib.h>
+#include <stdio.h>
 #include <wlr/render/interface.h>
 #include <wlr/render/pixman.h>
 #include <wlr/render/wlr_renderer.h>
@@ -211,6 +212,18 @@ bool wlr_renderer_init_wl_display(struct wlr_renderer *r,
 		return false;
 	}
 
+	char *str_formats = malloc(len * 11 + 1);
+	if (str_formats == NULL) {
+		return false;
+	}
+	for (size_t i = 0; i < len; i++) {
+		snprintf(&str_formats[i*11], (len - i) * 11 + 1, "0x%X ",
+			(unsigned int)formats[i]);
+	}
+	wlr_log(WLR_DEBUG, "Supported SHM formats: %s", str_formats);
+	// [DEBUG] [wlr] [render/wlr_renderer.c:223]
+	// Supported SHM formats: XB24 AB24 BG24 RX12 RA12 RX15 RA15 RG16 XB4H AB4H
+	// Supported SHM formats: 0x34324258 0x34324241 0x34324742 0x32315852 0x32314152 0x35315852 0x35314152 0x36314752 0x48344258 0x48344241 
 	bool argb8888 = false, xrgb8888 = false;
 	for (size_t i = 0; i < len; ++i) {
 		// ARGB8888 and XRGB8888 must be supported and are implicitly
-- 
2.33.1

