From 5af09bfd372ce77622f17f4e4e3f49a20516c369 Mon Sep 17 00:00:00 2001
From: this-user <this-user@users.noreply.github.com>
Date: Wed, 14 Aug 2019 16:41:56 +0200
Subject: [PATCH] Made compatible with 'Arrow' 0.14.5

---
 shreddit/shredder.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/shreddit/shredder.py b/shreddit/shredder.py
index c4633a0..ea20993 100644
--- a/shreddit/shredder.py
+++ b/shreddit/shredder.py
@@ -30,8 +30,8 @@ def __init__(self, config, user):
         if self._save_directory:
             self._r.config.store_json_result = True
 
-        self._recent_cutoff = arrow.now().replace(hours=-self._hours)
-        self._nuke_cutoff = arrow.now().replace(hours=-self._nuke_hours)
+        self._recent_cutoff = arrow.now().shift(hours=-self._hours)
+        self._nuke_cutoff = arrow.now().shift(hours=-self._nuke_hours)
         if self._save_directory:
             if not os.path.exists(self._save_directory):
                 os.makedirs(self._save_directory)
